frontend/src/routes/manage/Video/Access.tsx

```diff
@@ -0,0 +1,890 @@
+    };
+
+    const submit = async (acl: ACL) => {
+        // TODO: Actually save new ACL.
+        // eslint-disable-next-line no-console
+        console.log(acl);
+    };
+
+    return <div css={{ display: "flex", gap: 8, alignSelf: "flex-start", marginTop: 40 }}>
+        {/* Reset button */}
+        <ButtonWithModal
+            buttonKind="danger"
+            modalRef={resetModalRef}
+            label={t("manage.access.reset-modal.label")}
+            title={t("manage.access.reset-modal.title")}
+            body={t("manage.access.reset-modal.body")}
+            confirmationLabel={t("manage.access.reset-modal.label")}
+            handleClick={() => currentRef(resetModalRef).open()}
+            onConfirm={() => aclSelectRef.current?.reset?.()}
+        />
+        {/* Save button */}
+        <ButtonWithModal
+            buttonKind="happy"
+            modalRef={saveModalRef}
+            label={t("save")}
+            title={t("manage.access.save-modal.title")}
+            body={t("manage.access.save-modal.body")}
+            confirmationLabel={t("manage.access.save-modal.confirm")}
+            handleClick={() => {
+                const newACL = currentRef(aclSelectRef).selections();
+                return !containsUser(newACL) ? currentRef(saveModalRef).open() : submit(newACL);
+            }}
+            onConfirm={() => submit(currentRef(aclSelectRef).selections())}
+        />
+    </div>;
+};
+
+type ButtonWithModal = {
+    buttonKind: ButtonKind;
+    modalRef: RefObject<ModalHandle>;
+    label: string;
+    title: string;
+    body: string;
+    confirmationLabel: string;
+    handleClick: () => void;
+    onConfirm: () => void;
+}
+
+const ButtonWithModal: React.FC<ButtonWithModal> = ({ ...props }) => {
+    const { t } = useTranslation();
+    return <>
+        <Button
+            kind={props.buttonKind}
+            onClick={props.handleClick}
+        >{props.label}</Button>
+        <Modal ref={props.modalRef} title={props.title}>
+            <p>{props.body}</p>
+            <div css={{
+                display: "flex",
+                gap: 12,
+                justifyContent: "center",
+                flexWrap: "wrap",
+                marginTop: 32,
+            }}>
+                <Button onClick={() => currentRef(props.modalRef).close?.()}>
+                    {t("cancel")}
+                </Button>
+                <Button kind="danger" onClick={() => {
+                    props.onConfirm();
+                    currentRef(props.modalRef).close?.();
+                }}>{props.confirmationLabel}</Button>
+            </div>
+        </Modal>
+    </>;
+};
+
+type ACLSelectWrapper = {
+    initialACL: ACL;
+    userRequired?: boolean;
+}
+
+export type ACLWrapperHandle = {
+    selections: () => ACL;
+    reset?: () => void;
+};
+
+const UserRequiredContext = createContext<boolean>(false);
+
+export const ACLSelectWrapper = forwardRef<ACLWrapperHandle, ACLSelectWrapper>(
```

Comment:
I would probably rename this. It is the main entry point basically for the "ACL UI" right? The name sounds not "grand" enough for that IMO :D Especially this "wrapper". `ACLSelector`? 

Oh and regarding casing... `AclSelector`? At least in Rust, acronyms are not fully uppercased. Dunno how we have done it on the frontend side so far...